access_control.py

```diff
@@ -0,0 +1,225 @@
+                print(f"Processing feedstock for access control: {feedstock_repo}")
+                _process_request_for_feedstock(
+                    feedstock_repo, resource, remove=remove, policy_args=policy_args
+                )
+                action = "remove" if remove else "add"
+                update_access_yaml(resource, feedstock_repo, action=action)
+
+
+def process_access_control_requests():
+    """Process access control requests"""
+    print("Processing access control request")
+    _process_access_control_requests("grant_access", remove=False)
+    _process_access_control_requests("revoke_access")
+
+
+def _process_request_for_feedstock(feedstock, resource, remove, policy_args):
+    feedstock_clone_path = tempfile.mkdtemp()
+    assert GH_ORG
+    clone_cmd = f"git clone --depth 1 https://github.com/{GH_ORG}/{feedstock}.git {feedstock_clone_path}"
+    print(f"Cloning: {clone_cmd}")
+    subprocess.run(
+        clone_cmd,
+        shell=True,
+    )
+
+    register_ci_cmd = [
+        "conda-smithy register-ci",
+        f"--organization {GH_ORG}",
+        "--without-azure",
+        "--without-travis",
+        "--without-circle",
+        "--without-appveyor",
+        "--without-drone",
+        "--without-webservice",
+        "--without-anaconda-token",
+        f"--feedstock_directory {feedstock_clone_path}",
+        f"--cirun-resources {resource}",
+    ]
+
+    if policy_args:
+        policy_args_param = [f"--cirun-policy-args {arg}" for arg in policy_args]
+        register_ci_cmd.extend(policy_args_param)
+    if remove:
+        register_ci_cmd.append("--remove")
+
+    register_ci_cmd_str = " ".join(register_ci_cmd)
+    print(f"RegisterCI command: {register_ci_cmd_str}")
+    subprocess.check_call(register_ci_cmd_str,
+        shell=True,
+    )
+
+
+def check_if_repo_exists(feedstock_name):
+    repo = f"{feedstock_name}-feedstock"
+    owner_repo = f"{GH_ORG}/{repo}"
+    print(f"Checking if {owner_repo} exists")
+    response = requests.get(f"https://api.github.com/repos/{owner_repo}")
+    if response.status_code != 200:
+        raise ValueError(f"Repository: {owner_repo} not found!")
+
+def _check_for_path(path):
+    mapping = _get_filename_feedstock_mapping(path)
+    for filename, feedstocks in mapping.items():
+        print(f"Checking if {filename} is in {CIRUN_FILENAME_RESOURCE_MAPPING.keys()}")
+        assert filename in CIRUN_FILENAME_RESOURCE_MAPPING
+        for feedstock in feedstocks:
+            check_if_repo_exists(feedstock)
+    if not mapping:
+        print(f"Nothing to check for: {path}")
+
+def check():
+    """Check requests are valid (resources exist, feedstocks exist)"""
+    _check_for_path("grant_access")
+    _check_for_path("revoke_access")
+
+def _commit_after_files_removal(push=True):
+    subprocess.check_call(
+        "git add .",
+        shell=True,
+    )
+
+    subprocess.check_call(
+            "git commit --allow-empty -am 'Remove access control files'",
```

Comment:
What's the reason behind --allow-empty? Are we expecting no files to be added? Can we check for that and save the commit just in case we don't have to do anything?

Reply:
I think accidentally left the `allow-empty`